rpcs3/Emu/VFS.cpp

```diff
@@ -701,18 +705,75 @@ std::string vfs::unescape(std::string_view name)
 	return result;
 }
 
-bool vfs::host::rename(const std::string& from, const std::string& to, bool overwrite)
+bool vfs::host::rename(const std::string& from, const std::string& to, const lv2_fs_mount_point* mp, bool overwrite)
 {
-	while (!fs::rename(from, to, overwrite))
+	// Try the easy way
+	for (int i = 0; i < 5; i++, std::this_thread::yield())
 	{
-		// Try to ignore access error in order to prevent spurious failure
-		if (Emu.IsStopped() || fs::g_tls_error != fs::error::acces)
+		if (fs::rename(from, to, overwrite))
+		{
+			return true;
+		}
+
+		if (fs::g_tls_error != fs::error::acces)
 		{
 			return false;
 		}
 	}
 
-	return true;
+#ifdef _WIN32
+	// Hard way: lock mount point, close file descriptors, retry 
+	const auto from0 = from.substr(0, from.find_last_not_of("/\\") + 1) + '/';
+
+	// Lock app_home as well because it could be in the same drive as current mount point
+	std::scoped_lock lock(mp->mutex, g_mp_sys_app_home.mutex);
+
+	idm::select<lv2_fs_object, lv2_file>([&](u32 id, lv2_file& file)
+	{
+		if (vfs::get(file.name.data()).starts_with(from0))
+		{
+			verify(HERE), file.mp == mp || file.mp == &g_mp_sys_app_home;
+			file.restore_data.seek_pos = file.file.pos();
+			file.file.close(); // Actually close it!
+		}
+	});
+
+	bool res = false;
+
+	for (;; std::this_thread::yield())
+	{
+		if (fs::rename(from, to, overwrite))
+		{
+			res = true;
+			break;
+		}
+
+		if (Emu.IsStopped() || fs::g_tls_error != fs::error::acces)
+		{
+			res = false;
+			break;
+		}
+	};
+
+	const auto fs_error = fs::g_tls_error;
+
+	idm::select<lv2_fs_object, lv2_file>([&](u32 id, lv2_file& file)
+	{
+		if (auto name = vfs::get(file.name.data()); name.starts_with(from0))
+		{
+			// Reopen with ignored TRUNC, APPEND, CREATE and EXCL flags
+			auto res0 = lv2_file::open_raw((res ? to : from0) + '/' + name.substr(from0.size()), file.flags & CELL_FS_O_ACCMODE, file.mode, file.type, file.mp);
+			file.file = std::move(res0.file);
+			verify(HERE), file.file.operator bool();
```

Comment:
Note to self: this is actually a valid case and may happen because of previous file renaming, need to handle as well.

Reply:
Fixed!